src/NUnitFramework/framework/Attributes/TestCaseSourceAttribute.cs

```diff
@@ -213,7 +231,7 @@ private IEnumerable GetTestCaseSource(IMethodInfo method)
 
             // Handle Type implementing IEnumerable separately
             if (SourceName == null)
-                return Reflect.Construct(sourceType) as IEnumerable;
+                return Reflect.Construct(sourceType, null) as IEnumerable;
```

Comment:
So, your intention is that the third form should not support arguments? That's OK for now at least (till folks complain) but I want to get it straight for the documentation.